### DISCUSSION TITLE
IntelliJ IDEs Plugin: Static compilation is recommended.

### DISCUSSION THREAD

**Discussion by VergeDX** [community/community#17898] [Copilot Conversations]
I noticed the newer version "GitHub Copilot - IntelliJ IDEs Plugin" (`1.1.24.1610` for me)  
shows the fail-fast error "Failed to initiate the GitHub login process. Please try again."

Then I startup IDEs in the terminal and check out the JVM error logs, it shows me a file  
`~/.local/share/JetBrains/IntelliJIdea2022.1/github-copilot-intellij/copilot-agent/bin/copilot-agent-linux`
cannot execute due to an error "The file does not exist or could not be executed.".

Here is the output of comamnd `ldd ~/.local/share/JetBrains/IntelliJIdea2022.1/github-copilot-intellij/copilot-agent/bin/copilot-agent-linux`:

```````shell
~
at 11:54:02 fsh ❯ ldd ~/.local/share/JetBrains/IntelliJIdea2022.1/github-copilot-intellij/copilot-agent/bin/copilot-agent-linux
	linux-vdso.so.1 (0x00007ffc4a4f4000)
	libdl.so.2 => /nix/store/lyl6nysc3i3aqhj6shizjgj0ibnf1pvg-glibc-2.34-210/lib/libdl.so.2 (0x00007fda24b2b000)
	libstdc++.so.6 => not found
	libm.so.6 => /nix/store/lyl6nysc3i3aqhj6shizjgj0ibnf1pvg-glibc-2.34-210/lib/libm.so.6 (0x00007fda24a52000)
	libgcc_s.so.1 => /nix/store/lyl6nysc3i3aqhj6shizjgj0ibnf1pvg-glibc-2.34-210/lib/libgcc_s.so.1 (0x00007fda24a38000)
	libpthread.so.0 => /nix/store/lyl6nysc3i3aqhj6shizjgj0ibnf1pvg-glibc-2.34-210/lib/libpthread.so.0 (0x00007fda24a33000)
	libc.so.6 => /nix/store/lyl6nysc3i3aqhj6shizjgj0ibnf1pvg-glibc-2.34-210/lib/libc.so.6 (0x00007fda24833000)
	/lib64/ld-linux-x86-64.so.2 => /nix/store/lyl6nysc3i3aqhj6shizjgj0ibnf1pvg-glibc-2.34-210/lib64/ld-linux-x86-64.so.2 (0x00007fda24b32000)

```````

In view of I'm using a non-FHS Linux distribution NixOS, the error is pretty common.
"Downloading and attempting to run a binary on NixOS will almost never work.", from below link:
[https://nixos.wiki/wiki/Packaging/Binaries](https://nixos.wiki/wiki/Packaging/Binaries)

So we should fixing binary manully by using something link [https://github.com/NixOS/patchelf](https://github.com/NixOS/patchelf),  
or GitHub choice release the binary in static linked format, which also improved the compatbility.

**Comment by VergeDX** [comment]
If someone on NixOS wants to use it, here is a simple wrapper:

```````Nix
{ buildFHSUserEnv, ... }:
let src = ./.; in
let bin = "/bin/copilot-agent-linux"; in
buildFHSUserEnv rec {
  name = "github-copilot-intellij";
  runScript = "${src}/copilot-agent/${bin}.bak";
}
```````

I'm in `~/.local/share/JetBrains/IntelliJIdea2022.1/github-copilot-intellij` dir,  
and run `nix-build -E '(import <nixpkgs> { system = "x86_64-linux"; }).pkgs.callPackage ./. { }'`.  
Then backup `copilot-agent-linux` binary and replace with the script of result.

Due to https://github.com/github-community/community/discussions/16230, it still cannot work for me : (

**Comment by VergeDX** [comment]
See https://github.com/Vanilla-s-Lab/flakes/commit/031f252e8a8d13a39eaa1032df66063405b06622.  
Issues closed.

**Comment by hacker1024** [comment]
This can be done by patching the binary, no FHS required.

https://github.com/NixOS/nixpkgs/pull/201518

(@VergeDX)
